clang/lib/AST/Type.cpp

```diff
@@ -3907,6 +3923,31 @@ CountAttributedType::CountAttributedType(
     DeclSlot[i] = CoupledDecls[i];
 }
 
+StringRef CountAttributedType::getAttributeName(bool WithMacroPrefix) const {
+// TODO: This method isn't really ideal because it doesn't return the spelling
+// of the attribute that was used in the user's code. This method is used for
+// diagnostics so the fact it doesn't use the spelling of the attribute in
+// the user's code could be confusing (#113585).
+#define ENUMERATE_ATTRS(PREFIX)                                                \
```

Comment:
Lambda instead of a macro?

Reply:
I did this way because it avoids allocating anything because they are all constant strings. This is probably premature optimization though. I'll try to rewrite using a lambda.

Reply:
Hmm actually I think I would need support for `constexpr` `std::string` to implement this and still be able to return a `StringRef`. I think LLVM is still on c++17 so I'm not sure using a lambba is do-abe here.

Reply:
Yeah, the string literal concatenation kind more or less forces this to be a macro from what I can tell, and the underlying problem is that we don’t have access to the actual attribute, but fixing that issue is a bit tricky unfortunately...

Reply:
> Hmm actually I think I would need support for `constexpr` `std::string` to implement this

(fyi, even `constexpr std::string` wouldn’t be enough; an implementation of this that uses `std::string` would require non-transient `constexpr` allocation, which is something people are working on, but it’s not in the standard yet and no compiler supports anything like it afaik)